src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceHelpers.cs

```diff
@@ -51,18 +51,20 @@ public static string GetAssemblyDisplay(Compilation compilation, IAssemblySymbol
             // This method is only used to generate a comment at the top of Metadata-as-Source documents and
             // previous submissions are never viewed as metadata (i.e. we always have compilations) so there's no
             // need to consume compilation.ScriptCompilationInfo.PreviousScriptCompilation.
+            var assemblyReference = TryGetAssemblyReference(compilation, assemblySymbol);
+            return assemblyReference?.Display ?? FeaturesResources.location_unknown;
+        }
 
-            // TODO (https://github.com/dotnet/roslyn/issues/6859): compilation.GetMetadataReference(assemblySymbol)?
-            var assemblyReference = compilation.References.Where(r =>
+        public static MetadataReference TryGetAssemblyReference(Compilation compilation, IAssemblySymbol assemblySymbol)
```

Comment:
So this implies that IAssemblySymbol didn't come from the Compilation? Where *did* it come from?

Reply:
It was originally resolved in the context where the user was working, and here we are in the context of the temporary compilation that hosts the rendered MAS content.

Reply:
Fixed to use the correct compilation